### PR TITLE
[SPARK-34155][SQL][TEST] Add partition columns for TPCDS tables

### PR DESCRIPTION
**Pull request by wangyum**
### What changes were proposed in this pull request?

This pr add partition columns for TPCDS tables. The partition column is consistent with the [TPCDSTables](https://github.com/databricks/spark-sql-perf/blob/master/src/main/scala/com/databricks/spark/sql/perf/tpcds/TPCDSTables.scala).

### Why are the changes needed?

Better track plan changes. For example, [this is the change](https://github.com/apache/spark/commit/3fe1a93a4038a74f2d987ff8a74a90c7f4749a48) after SPARK-34119.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

N/A

### CODE REVIEW

**Comment by wangyum** [comment]
The actual change is: 
```diff
diff --git a/sql/core/src/test/scala/org/apache/spark/sql/TPCDSBase.scala b/sql/core/src/test/scala/org/apache/spark/sql/TPCDSBase.scala
index 9f420cac97..2fe79b273d 100644
--- a/sql/core/src/test/scala/org/apache/spark/sql/TPCDSBase.scala
+++ b/sql/core/src/test/scala/org/apache/spark/sql/TPCDSBase.scala
@@ -271,6 +271,41 @@ trait TPCDSBase extends SharedSparkSession {
       """.stripMargin
   )
 
+  private val tablePartitionColumns = Map(
+    "catalog_sales" -> Seq("cs_sold_date_sk"),
+    "catalog_returns" -> Seq("cr_returned_date_sk"),
+    "inventory" -> Seq("inv_date_sk"),
+    "store_sales" -> Seq("ss_sold_date_sk"),
+    "store_returns" -> Seq("sr_returned_date_sk"),
+    "web_sales" -> Seq("ws_sold_date_sk"),
+    "web_returns" -> Seq("wr_returned_date_sk"),
+    "call_center" -> Seq.empty[String],
+    "catalog_page" -> Seq.empty[String],
+    "customer" -> Seq.empty[String],
+    "customer_address" -> Seq.empty[String],
+    "customer_demographics" -> Seq.empty[String],
+    "date_dim" -> Seq.empty[String],
+    "household_demographics" -> Seq.empty[String],
+    "income_band" -> Seq.empty[String],
+    "item" -> Seq.empty[String],
+    "promotion" -> Seq.empty[String],
+    "reason" -> Seq.empty[String],
+    "ship_mode" -> Seq.empty[String],
+    "store" -> Seq.empty[String],
+    "time_dim" -> Seq.empty[String],
+    "warehouse" -> Seq.empty[String],
+    "web_page" -> Seq.empty[String],
+    "web_site" -> Seq.empty[String]
+  )
+
+  private def partitionedByClause(tableName: String) = {
+    if (tablePartitionColumns(tableName).nonEmpty) {
+      s"PARTITIONED BY (${tablePartitionColumns(tableName).mkString(", ")})"
+    } else {
+      ""
+    }
+  }
+
   val tableNames: Iterable[String] = tableColumns.keys
 
   def createTable(
@@ -282,6 +317,7 @@ trait TPCDSBase extends SharedSparkSession {
       s"""
          |CREATE TABLE `$tableName` (${tableColumns(tableName)})
          |USING $format
+         |${partitionedByClause(tableName)}
          |${options.mkString("\n")}
        """.stripMargin)
   }

```

**Comment by wangyum** [comment]
retest this please

**Comment by wangyum** [comment]
retest this please

**Comment by HyukjinKwon** [comment]
cc @maropu and @peter-toth too

File: sql/core/src/test/scala/org/apache/spark/sql/TPCDSBase.scala [hunk 1]
```diff
@@ -271,6 +271,23 @@ trait TPCDSBase extends SharedSparkSession {
       """.stripMargin
   )
 
+  private val tablePartitionColumns = Map(
```
Comment (maropu) [inline sql/core/src/test/scala/org/apache/spark/sql/TPCDSBase.scala:274]: nit: Could you leave some comments about how we decided these partition keys?

**Review by maropu — APPROVED** [review]

File: sql/core/src/test/scala/org/apache/spark/sql/TPCDSBase.scala (hunk 1 above)
Comment (wangyum) [reply sql/core/src/test/scala/org/apache/spark/sql/TPCDSBase.scala:275]: ok

**Comment by peter-toth** [comment]
LGTM, PlanStabilitySuite should work on partitioned tables

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**